### ISSUE TITLE
 Implement an API layer to support other frameworks such as Cutter and Binary Ninja

### ISSUE THREAD

**Issue by ITAYC0HEN** [VirusTotal/vt-ida-plugin#6]
Currently, the Plugin is built above IDA-Python and dependent on IDA to achieve its information. That, when IDA is basically supplying for the PLugin the information to work on, which is not special for IDA.

My suggestion is to use a more flexible and modern design for the Plugin in such a way that an API layer is implemented above CORE functionality. The logic will be implemented without being dependent on a specific disassembler to supply the information.

In this way, the community would be able to implement plugins for other disassemblers such as [Cutter](https://github.com/radareorg/cutter), Binary Ninja, GHIDRA, and others.

This method will expand the usability of the Plugin and your great implementation for IDA can be a template or a go-to reference for other plugins.

To sum up, the solution I would like to see will be

```
Cutter        IDA      Binja      GHIDRA
    |__________|___________|_________|
                    |
                API Layer
                    |
            VT Querying Logic
```

**Comment by gerardofn** [comment]
Thanks for your feedback.

I will work on a new abstraction layer for the next release.

**Comment by gerardofn** [comment]
v0.9 implements a new abstraction layer to enhance portability.
